### ISSUE TITLE
Nested class, interface or enum shouldn't be publicly visible

### ISSUE THREAD

**Issue by schalkms** [arturbosch/detekt#444]
Nested types are often used for implementing private functionality. Therefore, they shouldn't be externally visible.

implementation idea:

- take a look at the rules in the **detekt-rules** package
- implement `visitClass`
- visit nested classes
- check if class `isPublic`

example:
```kotlin
internal class ParentType {
  public class NestedType { // violation
  }
}
```

**Comment by tagantroy** [comment]
@schalkms @arturbosch can we close this issue?
because it's already implemented

**Comment by Mauin** [comment]
Thanks for the reminder @tagantroy!
